### ISSUE TITLE
Could you publish new apk on github?

### ISSUE THREAD

**Issue by linsui** [b-lam/Resplash#83]
I get this great app from @IzzySoft 's f-droid repo which picks apk from github release. There is no new release since version 1.3.3 . Would you like to publish new release on github? It will be useful for those don't or can't use google playstore. Thanks anyway!

**Comment by IzzySoft** [comment]
Did you give up? Or just misread the `2.0.1` in my repo comparing it to the `2.1.0` in `build.gradle`? I just wonder where that comes from. According to metadata, it must have been a tag which now is deleted but obviously existed in May 2019.

**Comment by b-lam** [comment]
@IzzySoft @linsui Sorry, I haven't had much time to get to this but I'm currently rewriting Resplash from scratch and should hopefully have an updated release soon.

**Comment by linsui** [comment]
Thanks! Great news!

**Comment by b-lam** [comment]
A new apk has been uploaded for v3.0.0

**Comment by IzzySoft** [comment]
Thanks! Will show up in my repo tomorrow. Wow, and even got a bit slimmer, 1M less in size – just kick Crashlytics as well (it's phased out anyway) for another slimming, and you'll be below 6M again :smiley:

**Comment by b-lam** [comment]
@IzzySoft Yup the rewrite definitely helped remove a bunch of unnecessary code and assets. Is Crashlytics being phased out? I think it is just Fabric Crashlytics that is being phased out for Firebase Crashlytics, which is what I have switched to.

**Comment by IzzySoft** [comment]
Yes, that. I didn't check which one you had implemented. To me both are just proprietary trackers collecting user data on "other people's computers", sorry (well, I don't like my data stored with the "big collectors" like Google etc – no offense meant of course). Consider switching to some FOSS analytics instead? :smiley: There're e.g. [ACRA](https://github.com/ACRA/acra), [tracedroid](https://github.com/ligi/tracedroid) and probably more…

**Comment by LongJohn-Silver** [comment]
What about publishing directly on F-droid? Any ideas about that?

**Comment by IzzySoft** [comment]
@LongJohn-Silver not possible before Crashlytics was removed. F-Droid doesn't permit proprietary elements. The following things currently used are show-stoppers for a listing at F-Droid:

```gradle
apply plugin: 'com.google.android.gms.oss-licenses-plugin'
apply plugin: 'com.google.firebase.crashlytics'
apply plugin: 'com.google.gms.google-services'
```

Before irritated questions arise: the first (oss-licenses-plugin) depends on the last (GMS), hence it is a show-stopper itself. Next:

```gradle
    implementation 'com.android.billingclient:billing:3.0.0'
```

Again depends on GMS.

```gradle
    // Firebase
    implementation 'com.google.firebase:firebase-analytics-ktx:17.5.0'
    implementation 'com.google.firebase:firebase-crashlytics:17.2.1'
    implementation 'com.google.firebase:firebase-inappmessaging-ktx:19.1.1'
    implementation 'com.google.firebase:firebase-firestore-ktx:21.6.0'
    implementation 'com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava'

    // Play Core
    implementation 'com.google.android.play:core:1.8.0'
    implementation 'com.google.android.play:core-ktx:1.8.1'
```

All proprietary, and thus show-stoppers.

**TL;DR:** A listing at "F-Droid proper" isn't that easily achieved here, and would need quite some work. Not sure if @b-lam is willing to do that (can be done via build flavors as long as the core functionality of the app doesn't depend upon one of those).

**Comment by b-lam** [comment]
@IzzySoft Thanks for the write-up! Removing these shouldn't affect the core functionality but unfortunately I don't really have the time or resources to support both versions right now. @LongJohn-Silver All the code is open source so you could fork this project and make it your own or submit a PR if this is something you really want.

**Comment by LongJohn-Silver** [comment]
> @IzzySoft Thanks for the write-up! Removing these shouldn't affect the core functionality but unfortunately I don't really have the time or resources to support both versions right now. @LongJohn-Silver All the code is open source so you could fork this project and make it your own or submit a PR if this is something you really want.

I am just learning to code on Python, so I don't think I can help here. But if I ever do, I'll be glad.

**Comment by IzzySoft** [comment]
@b-lam that's more than I dared hope for, thanks! May I suggest opening a corresponding issue with the label "help-needed"? Then those interested can jump in, fork, and provide the PR(s). You could also specify your preference, like

* I don't care, so just kick the crap out :smile:
* please split it off using build flavors named `fdroid` and `play` (or whatever names you prefer)
* mixed mode: just ditch Ⓐ but split off flavors for Ⓑ

Such an issue opened by you (the author and repo-owner) would carry some more weight than if someone else does it. It also eliminates the excuse "but I don't want to maintain a fork", as you explicitly state to accept (reasonable) PRs to integrate here – so after the merge, the contributor is "free" again (and the software remains free, of course :smile:)

My repo should show the progress on each new release following such a merge (as multiple elements are effected, there could be multiple PRs, potentially, each focusing on one lib or a group of libs). Feel free to ping me if you think I can give some advice or should take a look if it would be ready to try on F-Droid.org.

**Comment by LongJohn-Silver** [comment]
Also pinning said issue would create more attention.
